### PR TITLE
[code-infra] Closer sync with eslint config of codebase

### PR DESCRIPTION
**Pull request by oliviertassinari**
A step to simplify the eslint config in the codebase.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -94,15 +93,21 @@ module.exports = {
...
+        'material-ui/no-hardcoded-labels': 'off',
+      },
+    },
     {
       files: ['examples/**/*'],
       rules: {
         // We use it for demonstration purposes
         'no-console': 'off',
-        // Personal preference
-        'no-underscore-dangle': 'off',
```
Comment (oliviertassinari) [inline .eslintrc.js:105]: Seems marginally used

File: .eslintrc.js [hunk 2]
```diff
@@ -25,37 +50,11 @@ module.exports = {
...
-        patterns: [
-          {
-            group: ['lodash-es/*'],
-            message: 'Use `import { debounce } from "lodash-es";` instead.',
-          },
-        ],
-      },
-    ],
+    '@typescript-eslint/return-await': 'off',
+    'no-restricted-imports': ['error', noRestrictedImports],
```
Comment (oliviertassinari) [inline .eslintrc.js:57]: moved to a variable, `noRestrictedImports`, to be able to extend it in specific paths.

File: .eslintrc.js [hunk 3]
```diff
@@ -94,15 +93,21 @@ module.exports = {
         ],
       },
     ],
+    'material-ui/no-hardcoded-labels': 'off', // We are not really translating the docs/website anymore
   },
   overrides: [
+    ...baseline.overrides,
```
Comment (oliviertassinari) [inline .eslintrc.js:99]: The main difference

File: examples/with-prisma-data-provider/toolpad/prisma.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-underscore-dangle */
```
Comment (Janpot) [inline examples/with-prisma-data-provider/toolpad/prisma.ts:1]: In general we don't add eslint disable comments in examples, but turn off the rule in an eslint override.

Reason is that eslint is not used in the example, so it shouldn't contain eslint comments after I download it.

File: docs/data/toolpad/studio/components/list/List.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable filenames/match-exported */
 import * as React from 'react';
 import { List } from '@toolpad/studio-components';
 
+// TODO fix name
```
Comment (Janpot) [inline docs/data/toolpad/studio/components/list/List.js:5]: What's wrong with the name?

File: .eslintrc.js (hunk 1 above)
Comment (Janpot) [reply .eslintrc.js:105]: There is a reason. examples don't contain eslint. So the code shouldn't contain eslint comments.

**Comment by Janpot** [comment]
Did you see that this eslint is in the process of [deprecating](https://eslint.org/blog/2022/08/new-config-system-part-2/) its config format?

File: .eslintrc.js (hunk 1 above)
Comment (oliviertassinari) [reply .eslintrc.js:105]: This makes sense. Should we move this rule to the share eslint config then. It feels like we want this for all /examples folders through MUI codebase.

File: docs/data/toolpad/studio/components/list/List.js (hunk 5 above)
Comment (oliviertassinari) [reply docs/data/toolpad/studio/components/list/List.js:5]: filenames/match-exported rule wants the component to be called List but List is already in the scope.

I left the TODO to keep this a small time investment, my main objective was for new code to be less likely to bend the shared eslint rules. For sure, we can fix it in this PR too.

**Comment by oliviertassinari** [comment]
> The new format seems considerably easier to understand in terms of extending lint configs.

I briefly looked at it, but saw we are on ESLint v8 not v9 so ignored. I hated these files so much: https://github.com/mui/material-ui/blob/next/packages/api-docs-builder/.eslintrc.js. I'm happy it won't be possible anymore.

File: .eslintrc.js (hunk 1 above)
Comment (Janpot) [reply .eslintrc.js:105]: 👍 I'm good with that as well

File: docs/data/toolpad/studio/components/list/List.js (hunk 5 above)
Comment (Janpot) [reply docs/data/toolpad/studio/components/list/List.js]: I updated the naming everywhere.

**Review by Janpot — APPROVED** [review]
Fixed the naming and brought back the rule for examples. With clearer comment. Want to merge this as I see PRs coming in that would fail the new naming rules.

**Pull request merged**